api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -161,6 +161,13 @@
      */
     Attribute<Class<?>> TUPLE_CLASS = new Attribute<>(new Class2String<>());
 
+    /**
+     * Attribute of input port.
+     * This is a read-only attribute to query whether the input port is connected to a DelayOperator
+     * This is for iterative processing.
+     */
+    Attribute<Boolean> IS_CONNECTED_TO_DELAY_OPERATOR = new Attribute<>(false);
```

Comment:
@davidyan74 : There are some internal Attributes defined in LogicalPlan. I think we should do the same for this as well. Define it as an internal attribute and use that. Only thing is that you now will have to first check if portContext.getValue() returns null or not